docs: add ModuleWithProviders deprecation (#33266)
angular/angular: aio/content/guide/deprecations.md

| `@angular/common` | [`ReflectiveInjector`](#reflectiveinjector) | <!--v8--> v9 |
| `@angular/core` | [`CollectionChangeRecord`](#core) | <!--v7--> v9 |
| `@angular/core` | [`DefaultIterableDiffer`](#core) | <!--v7--> v9 |

> | `@angular/core` | [`ModuleWithProviders` without a generic](#moduleWithProviders) | v9 |

Reviewer: ```suggestion
| `@angular/core` | [`ModuleWithProviders` without a generic](#moduleWithProviders) | v10 |
```

| `@angular/router` | [`loadChildren` string syntax](#loadChildren) | v11 |
| `@angular/router` | [`ActivatedRoute` params and `queryParams` properties](#activatedroute-props) | unspecified |
| `@angular/core/testing` | [`TestBed.get`](#testing) | v12 |

> | `ModuleWithProviders` | [`ModuleWithProviders` without a generic](#moduleWithProviders) | v9 |

Reviewer: Should be moved up with the other `@angular/core` deprecations

```suggestion
| `@angular/core` | [`ModuleWithProviders` without a generic](#moduleWithProviders) | v9 |
```

| [`wtfLeave`](api/core/wtfLeave) | none | v8 | See [Web Tracing Framework](#wtf) |
| [`entryComponents`](api/core/NgModule#entryComponents) | none | v9 | See [`entryComponents`](#entryComponents) |
| [`ANALYZE_FOR_ENTRY_COMPONENTS`](api/core/ANALYZE_FOR_ENTRY_COMPONENTS) | none | v9 | See [`ANALYZE_FOR_ENTRY_COMPONENTS`](#entryComponents) |

> | [`ModuleWithProviders`](api/core/ModuleWithProviders) | none | v9 | See [`ModuleWithProviders`](#moduleWithProviders) |

Reviewer: Can you remove the diff from this section and add it to the "index" section here? https://github.com/angular/angular/blob/81f3d19c13cc9de593247bd83fce5a41d3886043/aio/content/guide/deprecations.md#index

Otherwise it looks like `ModuleWithProviders` itself is deprecated.  We want to be able to specify that it's only using `ModuleWithProviders` without a generic that's deprecated.

### `entryComponents` and `ANALYZE_FOR_ENTRY_COMPONENTS` no longer required
Previously, the `entryComponents` array in the `NgModule` definition was used to tell the compiler which components would be created and inserted dynamically. With Ivy, this isn't a requirement anymore and the `entryComponents` array can be removed from existing module declarations. The same applies to the `ANALYZE_FOR_ENTRY_COMPONENTS` injection token.

> ### `ModuleWithProviders` type without a generic

Reviewer 2: Should we add this to the table above under `@angular/core`?

Reviewer: +1 We should add a link to this section to the index up top

Some Angular libraries, such as `@angular/router` and `@ngrx/store`, implement APIs that return a type called `ModuleWithProviders` (typically via a method named `forRoot()`).
This type represents an `NgModule` along with additional providers.

> Angular version 9 deprecates use of `ModuleWithProviders` without an explicitly generic type, where the generic type refers to the type of the `NgModule`.

Reviewer: ```suggestion
Angular version 9 deprecates use of `ModuleWithProviders` without an explicitly generic type, where the generic type refers to the type of the `NgModule`. In a future version of Angular, the generic will no longer be optional.
```

> If you're using the CLI, `ng update` should [migrate your code automatically](guide/migration-module-with-providers).

Reviewer 2: Is this going to be added in a separate PR (/is this something you'd like me to contribute)?

Reviewer: Do you mean the guide? There's a PR here:

https://github.com/angular/angular/pull/33085

> If you're not using the CLI, you can make the change manually.

Reviewer: ```suggestion
If you're not using the CLI, you can add any missing generic types to your application manually. 
```

### Migrating `ModuleWithProviders`

 See the [dedicated migration guide for `ModuleWithProviders`](guide/migration-module-with-providers).

> Sengled Smart Wi-Fi LED Multicolor

Reviewer: Remove?
